lib/peer.ml

```diff
@@ -1,26 +1,24 @@
-type state =
+type status =
   | Alive
   | Suspicious
   | Faulty
 
 type t = {
-  address : string;
-  port : int;
-  known_peers : t list;
-  state : state;
+  address : Address.t;
+  mutable status : status;
+  mutable peers : (Address.t, t) Base.Hashtbl.t;
 }
 
-let from_sockaddr sockaddr =
-  let open Lwt_unix in
-  match sockaddr with
-  | ADDR_UNIX _ -> failwith "Unix socket addresses not supported"
-  | ADDR_INET (inet, port) ->
-    {
-      address = Unix.string_of_inet_addr inet;
-      port;
-      known_peers = [];
-      state = Alive;
-    }
+let retrieve_peer_from_address_opt (peers : (Address.t, t) Base.Hashtbl.t)
+    address =
+  Base.Hashtbl.find peers address
 
```

Comment:
Why not call it `of_address` so it reads `Peer.of_address client.address` for example

Reply:
Yeah, my comment got misplaced :sweat_smile: you're right, what I suggest doesn't make sense for where this comment ended up.